Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -4847,19 +4847,22 @@ AccessGenerationResult InlineCacheCompiler::compile(const GCSafeConcurrentJSLock
     CCallHelpers jit(codeBlock);
     m_jit = &jit;
 
+    if (ASSERT_ENABLED)
+        jit.move(CCallHelpers::stackPointerRegister, jit.scratchRegister());
+
+    m_preservedReusedRegisterState = allocator.preserveReusedRegistersByPushing(jit, ScratchRegisterAllocator::ExtraStackSpace::NoExtraSpace);
```

Comment:
This can clobber scratchRegister. So having this between scratchRegister and its use is not ok.

Reply:
I think it is getting worse for ARM64 as the assertion becomes more and more complicated. Please do jot change anything on 64bit platform.

Reply:
Would you prefer having an entirely separate assertion block for `CPU(ARM)` or simply disabling the assertion for `CPU(ARM)`?